core/sync/textile/textile_test.go

```diff
@@ -0,0 +1,77 @@
+package textile_test
+
+import (
+	"testing"
+
+	"github.com/FleekHQ/space-poc/core/sync/textile"
+	"github.com/FleekHQ/space-poc/core/textile/handler"
+	"github.com/FleekHQ/space-poc/mocks"
+	"github.com/stretchr/testify/mock"
+	tc "github.com/textileio/go-threads/api/client"
+)
+
+func TestTextileHandler_OnCreate(t *testing.T) {
+	n := new(mocks.TextileNotifier)
+	th := textile.NewHandler(n)
+
+	b := []byte(`{"Key":"bafzbeid2zp544qy6ktwdlr5xxsmsioclbxj42dkbqckm35e6l5biqlo3tq","Name":"test-bucket-1"}`)
+
+	buck := &handler.Bucket{}
+	action := tc.Action{
+		Collection: "buckets",
+		Type:       1,
+		InstanceID: "dummy-id",
+		Instance:   b,
+	}
+	evt := &tc.ListenEvent{
+		Action: action,
+	}
+
+	n.On("SendTextileEvent", mock.Anything).Return()
+
+	th.OnCreate(buck, evt)
+}
+
+func TestTextileHandler_OnRemove(t *testing.T) {
+	n := new(mocks.TextileNotifier)
+	th := textile.NewHandler(n)
+
+	b := []byte(`{"Key":"bafzbeid2zp544qy6ktwdlr5xxsmsioclbxj42dkbqckm35e6l5biqlo3tq","Name":"test-bucket-1"}`)
+
+	buck := &handler.Bucket{}
+	action := tc.Action{
+		Collection: "buckets",
+		Type:       1,
+		InstanceID: "dummy-id",
+		Instance:   b,
+	}
+	evt := &tc.ListenEvent{
+		Action: action,
+	}
+
+	n.On("SendTextileEvent", mock.Anything).Return()
+
+	th.OnRemove(buck, evt)
+}
+
+func TestTextileHandler_OnSave(t *testing.T) {
+	n := new(mocks.TextileNotifier)
+	th := textile.NewHandler(n)
+
+	b := []byte(`{"Key":"bafzbeid2zp544qy6ktwdlr5xxsmsioclbxj42dkbqckm35e6l5biqlo3tq","Name":"test-bucket-1"}`)
+
+	buck := &handler.Bucket{}
+	action := tc.Action{
+		Collection: "buckets",
+		Type:       1,
+		InstanceID: "dummy-id",
+		Instance:   b,
+	}
+	evt := &tc.ListenEvent{
+		Action: action,
+	}
+
+	n.On("SendTextileEvent", mock.Anything).Return()
+
+	th.OnSave(buck, evt)
```

Comment:
you can add assertions to mocks if needed

Reply:
Fixed in [89c3390](https://github.com/FleekHQ/space-poc/pull/41/commits/89c339021677ee2ac9af22f4a9d7d1700c08f8e7).